services/requester/key_store_component.go

```diff
@@ -0,0 +1,131 @@
+package requester
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/onflow/flow-evm-gateway/config"
+	"github.com/onflow/flow-go-sdk/access"
+	"github.com/onflow/flow-go-sdk/crypto"
+	"github.com/onflow/flow-go/module/component"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	"github.com/rs/zerolog"
+)
+
+type KeyStoreComponent struct {
+	*KeyStore
+
+	log              zerolog.Logger
+	config           config.Config
+	client           access.Client
+	startupCompleted chan struct{}
+}
+
+var _ component.Component = (*KeyStoreComponent)(nil)
+
+func NewKeyStoreComponent(log zerolog.Logger, config config.Config, client access.Client) *KeyStoreComponent {
+	ks := &KeyStoreComponent{
+		log:              log,
+		config:           config,
+		client:           client,
+		startupCompleted: make(chan struct{}),
+	}
+
+	return ks
+}
+
+func (k *KeyStoreComponent) Start(ctx irrecoverable.SignalerContext) {
+	defer close(k.startupCompleted)
+
+	k.log.Info().Msg("starting key store component")
+
+	accountKeys := make([]*AccountKey, 0)
+	account, err := k.client.GetAccount(ctx, k.config.COAAddress)
+	if err != nil {
+		ctx.Throw(fmt.Errorf(
+			"failed to get signer info account for address: %s, with: %w",
+			k.config.COAAddress,
+			err,
+		))
+
+		return
+	}
+	signer, err := createSigner(ctx, k.config, k.log)
+
+	if err != nil {
+		ctx.Throw(err)
+
+		return
+	}
+	for _, key := range account.Keys {
+		// Skip account keys that do not use the same Public Key as the
+		// configured crypto.Signer object.
+		if !key.PublicKey.Equals(signer.PublicKey()) {
+			continue
+		}
+
+		accountKeys = append(accountKeys, &AccountKey{
+			AccountKey: *key,
+			Address:    k.config.COAAddress,
+			Signer:     signer,
+		})
+	}
+
+	k.KeyStore = NewKeyStore(accountKeys)
+
+}
+
+func (k *KeyStoreComponent) Ready() <-chan struct{} {
+	ready := make(chan struct{})
+
+	go func() {
+		<-k.startupCompleted
+		close(ready)
+	}()
+
+	return ready
+}
+
+func (k *KeyStoreComponent) Done() <-chan struct{} {
+	done := make(chan struct{})
+
+	go func() {
+		<-k.startupCompleted
```

Comment:
Does this need to be a Component? looks like it's a better for for a module which is run one and completes